InfrastructureManager/agents/azure_agent.py

```diff
@@ -0,0 +1,454 @@
+                                          ' specified.'.format(param))
+
+  def open_connection(self, parameters):
+    """ Connects to Microsoft Azure with the given credentials, creates an
+    authentication token and uses that to get the ServicePrincipalCredentials
+    which is needed to access any resources.
+    Args:
+      parameters: A dict, containing all the parameters necessary to authenticate
+        this user with Azure. We assume that the user has already authorized this
+        account by creating a Service Principal with the appropriate (Contributor)
+        role.
+    Returns:
+      A ServicePrincipalCredentials instance, that can be used to access or
+        create any resources.
+    """
+    app_id = parameters[self.PARAM_APP_ID]
+    app_secret_key = parameters[self.PARAM_APP_SECRET] + "="
+    tenant_id = parameters[self.PARAM_TENANT_ID]
+
+    # Get an Authentication token using ADAL.
+    context = adal.AuthenticationContext(self.AZURE_AUTH_ENDPOINT + tenant_id)
+    token_response = context.acquire_token_with_client_credentials(
+      self.AZURE_RESOURCE_URL, app_id, app_secret_key)
+    token_response.get('accessToken')
+
+    # To access Azure resources for an application, we need a Service Principal
+    # with the accurate role assignment. It can be created using the Azure CLI.
+    credentials = ServicePrincipalCredentials(client_id=app_id,
+                                              secret=app_secret_key,
+                                              tenant=tenant_id)
+    return credentials
+
+  def create_network_interface(self, network_client, interface_name, ip_name,
+                               subnet, parameters):
+    """ Creates the Public IP Address resource and uses that to create the
+    Network Interface.
+    Args:
+      network_client: A NetworkManagementClient instance.
+      interface_name: The name to use for the Network Interface.
+      ip_name: The name to use for the Public IP Address.
+      subnet: The Subnet resource from the Virtual Network created.
+      parameters:  A dict, containing all the parameters necessary to
+        authenticate this user with Azure.
+    """
+    group_name = parameters[self.PARAM_RESOURCE_GROUP]
+    region = parameters[self.PARAM_ZONE]
+    utils.log("Creating/Updating the Public IP Address '{}'".format(ip_name))
+    ip_address = PublicIPAddress(
+      location=region, public_ip_allocation_method=IPAllocationMethod.dynamic,
+      idle_timeout_in_minutes=4)
+    result = network_client.public_ip_addresses.create_or_update(
+      group_name, ip_name, ip_address)
+    self.sleep_until_update_operation_done(result, ip_name)
+    public_ip_address = network_client.public_ip_addresses.get(group_name, ip_name)
+
+    utils.log("Creating/Updating the Network Interface '{}'".format(interface_name))
+    network_interface_ip_conf = NetworkInterfaceIPConfiguration(
+      name='default', private_ip_allocation_method=IPAllocationMethod.dynamic,
```

Comment:
Could someone choose a different name from their account and maybe use that?
What if an existing default network had conflicting network rules with what AppScale needs to function?